### PR TITLE
Add alias search fallback to character()

### LINKED ISSUE #11: `unicode_names2::character("BACKSPACE")` returns None
I'd like to check if this is intended design or a bug.

If it is an intended design, I'd like to know adding them as another new API to this crate is okay or it belongs to other crate's responsibility.

Thanks!

### PR DESCRIPTION
**Pull request by youknowone**
fix #11 

Because I couldn't easily control the vendored generator, I used external phf crate for the last commit.
If you don't like it, it can be replaced by other stuff using first 3 commits which actually populating alias data.

cc @DimitrisJim

### CODE REVIEW

File: generator/src/main.rs [hunk 1]
```diff
@@ -81,12 +89,21 @@ fn get_table_data() -> (Vec<(char, String)>, Vec<(char, char)>) {
                 } else {
                     panic!("unknown hangul syllable {}", name)
                 }
+<<<<<<< HEAD
+            } else {
+=======
+            } else if !alias.is_empty() {
+>>>>>>> 6ea912b (Add alias search fallback for character())
```
Comment (progval) [inline generator/src/main.rs]: unresolved merge conflict

**Comment by progval** [comment]
> Because I couldn't easily control the vendored generator

It's not vendored; feel free to change it

File: generator/src/main.rs (hunk 1 above)
Comment (youknowone) [reply generator/src/main.rs]: my bad, I ran test again before final push this time.

**Comment by youknowone** [comment]
When I ran `cargo run --release -- --phf`, it generated different `generated_phf` to current one.
I am not confident to add a new database not breaking it.

**Comment by progval** [comment]
Ah yeah, it's not deterministic

**Comment by progval** [comment]
Why is it an issue, though? You could make it generate to `src/generated_alias.rs` and not commit the change to `src/generated.rs`.

Though it probably makes sense to replace the built-in perfect-hash with the `phf` crate altogether in a different PR. (We just need to check it doesn't come with size of perf drawbacks).

**Comment by youknowone** [comment]
I didn't know it is not deterministic. I found db bugs here, so I will rewrite db part soon.

**Comment by youknowone** [comment]
ok, bug is fixed but still phf is not ported.

**Comment by youknowone** [comment]
rebase on master but still using phf crate

**Comment by progval** [comment]
@youknowone Looks like we're all done here. Am I forgetting anything?

**Comment by youknowone** [comment]
The feature is done.
You may want to rewrite this patch using the internal phf implementation instead of phf crate

**Pull request merged**